### ISSUE TITLE
exception is swollowed when it should not be 

### ISSUE THREAD

**Issue by ashkan-saeedi-mazdeh** [dotnet/orleans#2242]
I was serializing this struct by sending it from a silo to subscribed clients in a MS test based project using SMS streams.

``` csharp
public struct ChallengeEventData
    {
        public enum EventType { Request, Accept, Deny }
        public IPlayer otherPlayer { get; set; }
        public EventType Type { get; set; }
        public byte[] data { get; set; }
        public Guid gameID { get; set; }

        public override string ToString()
        {
            return $"{Type} {data} {otherPlayer} {gameID}";
        }
    }
```

Until I added the Serializable attribute so binary formatter could serialize it, when I called `OnNextAsync` if there was a subscriber which needed serialization, then it failed silently by logging the exception. I only understood when ran it as a console project and saw the log.

Two things,

1- How can we force serializers to be generated for a type in code-gen?
2- This exception should be rethrown after logging I guess right?

I discussed with @ReubenBond  and he was of the opinion that the exception should not be swallowed too.

**Comment by sergeybykov** [comment]
Did you configure SMS for fire-and-forget message delivery?

**Comment by ashkan-saeedi-mazdeh** [comment]
@sergeybykov  IDK, I just used AddMemoryStream in programmatic config and did not do any changes but if that is the default case, then It is obvious that it will not throw since it will not get hold of the task, I'm away from the code and got a bad cold so can not check what the default is but I'm pretty sure I only used the helper method on the config to add the SMS provider.

**Comment by sergeybykov** [comment]
Fire-and-forget is not the default. So if you are awaiting `OnNextAsync`, then you should see the exceptions. It that not the case?

**Comment by ReubenBond** [comment]
@ashkan-saeedi-mazdeh if you debug and break on First-Chance Exceptions, do you see a serialization error? I suspect it's that.

**Comment by ashkan-saeedi-mazdeh** [comment]
@ReubenBond  @sergeybykov  I don't have the code back then but I wasn't awaiting the send to get the exception IIRC.
Next weekend that I will work on the project again, will try to remake it a faulty one and test and report back. On a busy contract during the week for now But not a one man team anymore on this project soon, we added a part time dev for our web stuff and a guy will be added soon to work full time on this. :)

**Comment by ReubenBond** [comment]
@ashkan-saeedi-mazdeh this ought to have been fixed in #2364 - do you know if it's still an issue?

**Comment by ashkan-saeedi-mazdeh** [comment]
@ReubenBond  yes fixed
